pkg/kubernetes/client.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/pkg/log"
+	"istio.io/pkg/version"
+)
+
+// Client is a helper wrapper around the Kube RESTClient for istioctl -> Pilot/Envoy/Mesh related things
+type Client struct {
+	Config *rest.Config
+	*rest.RESTClient
+}
+
+// ExecClient is an interface for remote execution
+type ExecClient interface {
+	GetIstioVersions(namespace string) (*version.MeshInfo, error)
+	PodsForSelector(namespace, labelSelector string) (*v1.PodList, error)
+	ConfigMapForSelector(namespace, labelSelector string) (*v1.ConfigMapList, error)
+}
+
+// NewClient is the constructor for the client wrapper
+func NewClient(kubeconfig, configContext string) (*Client, error) {
```

Comment:
There's some duplication between here and pkg/manifest/installer.go

Reply:
It is not there yet, I will keep this func in client.go. I found other duplicate functions and dedup-ed them.
Also I removed kubernetes package and move client.go into installer package.